### PR TITLE
Replace deprecated `page` methods

### LINKED ISSUE #37832: Investigate the use of deprecated `page.click` and replace where appropriate with `locator.click()` instead
### Describe the solution you'd like

Whilst looking into the PR to solve https://github.com/woocommerce/woocommerce/pull/37831, it was noted that page.click is deprecated as noted in the [Playwright docs](https://playwright.dev/docs/api/class-page#page-click)

This task is to investigate the use of page.click and replace with the locator-based [locator.click()](https://playwright.dev/docs/api/class-locator#locator-click) instead

### Describe alternatives you've considered

None considered - this is replacing a deprecated method

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rodelgc**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/37832.

This PR updates all deprecated `page` methods like `page.click()` with more standard commands like `page.locator().click()`. It also includes minor changes like:

- Added missing `await` keywords in a few areas (this seems to be one cause of flakiness)
- Replaced `page.waitForLoadState` commands with web assertions in a few areas, as suggested in the Playwright docs. Replacing _all_ of them will be done on a separate task.
- Fixed several inaccurate assertions, and removed a few unnecessary ones I found along the way
- Removed the `create-variable-spec` to avoid conflicting with https://github.com/woocommerce/woocommerce/pull/38335

However, this PR does not change the locator arguments used. For example, in the code
```js
adminPage.fill( 'input[name="log"]', admin.username )
```
the argument `input[name="log"]` was kept during the update like so:
```js
adminPage.locator( 'input[name="log"]' ).fill( admin.username )
```

This PR tries its best not to make big changes to the locator arguments, only when necessary. Changing to more user-facing locators will be done on a different task.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Testing locally

1. Run the usual commands for setting up the local wp-env environment
1. `cd` into `plugins/woocommerce`
1. Run all E2E tests `pnmp test:e2e-pw`. Make sure that they pass.
2. Also feel free to search through the `e2e-pw` folder for any remaining deprecated methods, using VS Code or any IDE. The list of deprecated methods are found in https://playwright.dev/docs/api/class-locator#deprecated and https://playwright.dev/docs/api/class-page#deprecated. There should be none left.

#### Testing on CI
1. Rerun the E2E tests a few times on CI to make sure that no new flakiness was introduced.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/test-data/data.js [hunk 1]
```diff
@@ -102,9 +102,9 @@ const storeDetails = {
 			zip: 'Division 1',
 			email: admin.email,
 			country: 'Liberia — Margibi', // corresponding to the text value of the option,
-			countryCode: 'LR',
+			countryCode: 'LR:MA',
 		},
-		expectedNumberOfIndustries: 8, // There are 8 checkboxes on the page (in Liberia), adjust this constant if we change that
+		expectedNumberOfIndustries: 7, // There are 7 checkboxes on the page (in Liberia), adjust this constant if we change that
```
Comment (rodelgc) [inline plugins/woocommerce/tests/e2e-pw/test-data/data.js:107]: I encountered this issue during PR testing so I went ahead and included this minor fix.

**Comment by rodelgc** [comment]
I'll submit this PR for review when https://github.com/woocommerce/woocommerce/pull/38335 is merged.

**Comment by rodelgc** [comment]
Submitting this PR now for review. @nigeljamesstevenson this PR contains several changes to the texts used as locator arguments, and might therefore affect your PR (https://github.com/woocommerce/woocommerce/pull/38400). Just in case this PR gets merged before yours, I'd be happy to help with updating your PR if there are conflicts with mine.

**Comment by nigeljamesstevenson** [comment]
> Submitting this PR now for review. @nigeljamesstevenson this PR contains several changes to the texts used as locator arguments, and might therefore affect your PR (https://github.com/woocommerce/woocommerce/pull/38400). Just in case this PR gets merged before yours, I'd be happy to help with updating your PR if there are conflicts with mine.

This is great stuff - Thanks a lot @rodelgc ! 
I will test and merge your PRs and will work through any conflicts with my [PR](https://github.com/woocommerce/woocommerce/pull/38400) - any issues and I'll hit you up - thanks again 😊

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-shipping-zones.spec.js
Comment (nigeljamesstevenson) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/create-shipping-zones.spec.js:1]: There are still a fair few page.click calls on the following lines:

- 96
- 159
- 310
- 312
- 316
- 322
- 330
- 332
- 342

**Comment by nigeljamesstevenson** [comment]
There are a few page.click calls in the `create-variations.spec.js` on the following lines:

- 54
- 63
- 113

**Comment by nigeljamesstevenson** [comment]
There are page.dispatchEvent calls in the following locations:
e2e-pw/global-teardown.js line

- 32

merchant/create-shipping-classes.spec.js line

- 12
- 16
- 20

merchant/order-coupon.spec.js line

- 136 - although based on the comment in the code this one may require some thought

**Comment by nigeljamesstevenson** [comment]
There are page.fill calls in the following locations:

merchant/create-shipping-zones.spec.js at line

- 308
- 331

**Comment by nigeljamesstevenson** [comment]
There are page.hover calls in the following locations:

merchant/create-shipping-zones.spec.js line

- 335

**Comment by nigeljamesstevenson** [comment]
There are page.isVisible calls in the following locations:

merchant/create-shipping-zones.spec.js line

- 335

**Comment by nigeljamesstevenson** [comment]
There are page.selectOption calls in the following locations:

merchant/products/add-variable-product/create-variable-product.spec.js line

- 41

**Comment by nigeljamesstevenson** [comment]
There are page.textContent calls in the following locations:

activate-and-setup/complete-onboarding-wizard.spec.js line

- 53
- 76

admin-analytics/analytics-overview.spec.js line

- 29

merchant/order-refund.spec.js line

- 57

shopper/order-email-receiving.spec.js line

- 91

e2e-pw/utils/onboarding.js line

- 35
- 46
- 71
- 85
- 111
- 146

**Comment by nigeljamesstevenson** [comment]
There are page.type calls in the following locations:

merchant/create-shipping-zones.spec.js line

- 311

**Comment by nigeljamesstevenson** [comment]
There are page.waitForSelector calls in the following locations:

merchant/create-shipping-zones.spec.js line

- 336

**Review by nigeljamesstevenson — CHANGES_REQUESTED** [review]
Hi @rodelgc - great work making all these updates - this is a BIG task 😊

I have listed a few more that I have spotted -  I know some of these will be taken into consideration with variations etc, but have listed them all for completeness - thanks!

**Comment by rodelgc** [comment]
Thank you @nigeljamesstevenson 
Done addressing the review comments. Resubmitting for another round of review.

**Comment by nigeljamesstevenson** [comment]
Looks great @rodelgc - unfortunately a few conflicts to be resolved

**Review by nigeljamesstevenson — APPROVED** [review]

**Comment by rodelgc** [comment]
> Looks great @rodelgc - unfortunately a few conflicts to be resolved

Done resolving merge conflicts in my latest change (https://github.com/woocommerce/woocommerce/pull/38344/commits/7cec7fe70e16001979f3a73c1f525de7fd89a7f2)

There were only 3 files I changed (updated the deprecated locators) in that commit:
- complete-onboarding-wizard.spec.js
- payment.spec.js
- page-loads.spec.js

Merging this PR now. Thanks for the extensive review @nigeljamesstevenson

**Pull request merged**